src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,212 @@
+        String query = "SELECT c.id AS car_id, model, manufacturer_id, name, country "
+                + "FROM cars c INNER JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id "
+                + "JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "WHERE cd.driver_id = ? AND c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCarFromResultSet(resultSet);
+                List<Driver> drivers = getDriversFromCar(car.getId());
+                car.setDrivers(drivers);
+                cars.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't get cars from DB by the driver ID: " + driverId, e);
+        }
+        return cars;
+    }
+
+    private void addDriverCarRelations(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't add drivers: " + car.getDrivers()
+                            + " to the car with ID: " + car.getId(), e);
+        }
+    }
+
+    private void deleteDriverCarRelations(Long carId) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, carId);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't delete drivers for car with ID: " + carId, e);
+        }
+    }
+
+    private Car getCarFromResultSet(ResultSet resultSet) throws SQLException {
+        return new Car(
+                resultSet.getObject("car_id", Long.class),
+                resultSet.getString("model"),
+                new Manufacturer(
+                        resultSet.getObject("manufacturer_id", Long.class),
+                        resultSet.getString("name"),
+                        resultSet.getString("country")), null);
+    }
+
+    private List<Driver> getDriversFromCar(Long carId) {
+        String getAllDriversFromCarRequest = "SELECT id, name, lisense_number "
```

Comment:
not a mistake, but more idiomatic to use query in that context

Reply:
Sorry, I can't understand what you mean by the definition of "more idiomatic".
Respectively, what I had to correct?